src/engagement_db_to_analysis/code_imputation_functions.py

```diff
@@ -37,20 +39,25 @@ def _impute_age_category(user, messages_traced_data, analysis_dataset_configs):
     for analysis_dataset_config in analysis_dataset_configs:
         for coding_config in analysis_dataset_config.coding_configs:
             if coding_config.analysis_dataset == age_category_cc.age_category_config.age_analysis_dataset:
+
+                assert age_coding_config is None, f"Found more than one age_coding_config in analysis_dataset_config"
                 age_coding_config = coding_config
                 age_engagement_db_datasets = analysis_dataset_config.engagement_db_datasets
 
     # Check and impute age_category in age messages only
-    log.info(f"Imputing {age_category_cc.analysis_dataset} from {age_coding_config.analysis_dataset} messages...")
-    for message_index, message in enumerate(messages_traced_data):
+    log.info(f"Imputing {age_category_cc.analysis_dataset} labels from {age_coding_config.analysis_dataset} messages...")
+    imputed_labels = 0
+    age_messages = 1
+    for message in messages_traced_data:
         if message["dataset"] in age_engagement_db_datasets:
-            age_label = message["labels"][0]
-            age_code = age_coding_config.code_scheme.get_code_with_code_id(age_label["CodeID"])
+
+            age_labels = get_latest_labels_with_code_scheme(Message.from_dict(dict(message)), age_coding_config.code_scheme)
```

Comment:
Maybe you could deserialize earlier so you can say things like if message.dataset in age_engagement_db_datasets

Reply:
I prefer deserializing on-demand here as the format is not used anywhere else in this function.

Reply:
message["dataset"] is used on L52, and you get slightly better editor support with message.dataset